logstash-core/src/test/java/org/logstash/util/CATrustedFingerprintTrustStrategyTest.java

```diff
@@ -0,0 +1,110 @@
+    private static final Date DATE_CERTS_VALID = Date.from(DATE_CERTS_GENERATION.toInstant().plus(5, ChronoUnit.MINUTES));
+    private static final Date DATE_CERTS_EXPIRED = Date.from(DATE_CERTS_GENERATION.toInstant().plus(5000, ChronoUnit.DAYS));
+
+    @Before
+    public void puts() {
+        System.out.format("ROOT: `%s` (%s)\n", FINGERPRINT_ROOT, CERTIFICATE_ROOT.getSubjectX500Principal());
+        System.out.format("INTR: `%s` (%s)\n", FINGERPRINT_INTERMEDIATE, CERTIFICATE_INTERMEDIATE.getSubjectX500Principal());
+        System.out.format("SRVR: `%s` (%s)\n", FINGERPRINT_SERVER, CERTIFICATE_SERVER.getSubjectX500Principal());
+    }
+
+    @Test
+    public void testIsTrustedWhenAMatchingValidCertificateIsRootOfTheChain() throws CertificateException {
+        final CATrustedFingerprintTrustStrategy trustStrategy = new CATrustedFingerprintTrustStrategy(FINGERPRINT_ROOT, ()-> DATE_CERTS_VALID);
+        final X509Certificate[] chain = {CERTIFICATE_SERVER, CERTIFICATE_INTERMEDIATE, CERTIFICATE_ROOT};
+
+        assertTrue(trustStrategy.isTrusted(chain, "noop"));
+    }
+
+    @Test
+    public void testIsTrustedWhenAMatchingValidCertificateIsIntermediateOfTheChain() throws CertificateException {
+        final CATrustedFingerprintTrustStrategy trustStrategy = new CATrustedFingerprintTrustStrategy(FINGERPRINT_INTERMEDIATE, ()-> DATE_CERTS_VALID);
+        final X509Certificate[] chain = {CERTIFICATE_SERVER, CERTIFICATE_INTERMEDIATE, CERTIFICATE_ROOT};
+
+        assertTrue(trustStrategy.isTrusted(chain, "noop"));
+    }
+
+    @Test(expected = CertificateException.class)
+    public void testIsTrustedWhenAMatchingExpiredCertificateIsOnTheChain() throws CertificateException {
+        final CATrustedFingerprintTrustStrategy trustStrategy = new CATrustedFingerprintTrustStrategy(FINGERPRINT_ROOT, ()-> DATE_CERTS_EXPIRED);
+        final X509Certificate[] chain = {CERTIFICATE_SERVER, CERTIFICATE_INTERMEDIATE, CERTIFICATE_ROOT};
+
+        trustStrategy.isTrusted(chain, "noop");
+    }
+
+    @Test
+    public void testIsTrustedWhenNoMatchingCertificateIsOnTheChain() throws CertificateException {
+        final CATrustedFingerprintTrustStrategy trustStrategy = new CATrustedFingerprintTrustStrategy("abad1deaabad1deaabad1deaabad1deaabad1deaabad1deaabad1deaabad1dea", ()-> DATE_CERTS_VALID);
+        final X509Certificate[] chain = {CERTIFICATE_SERVER, CERTIFICATE_INTERMEDIATE, CERTIFICATE_ROOT};
+
+        assertFalse(trustStrategy.isTrusted(chain, "noop"));
+    }
+
+    @Test(expected = CertificateException.class)
+    public void testIsTrustedWhenDisjointedChainPresented() throws CertificateException {
+        final CATrustedFingerprintTrustStrategy trustStrategy = new CATrustedFingerprintTrustStrategy(FINGERPRINT_ROOT, ()-> DATE_CERTS_VALID);
+        final X509Certificate[] chain = {CERTIFICATE_SERVER, CERTIFICATE_ROOT};
+
+        assertFalse(trustStrategy.isTrusted(chain, "noop"));
```

Comment:
nit: remove unreachable assertion

```suggestion
        trustStrategy.isTrusted(chain, "noop");
```